ts/webui/src/components/common/ExpandableDetails/OpenRow.tsx

```diff
@@ -0,0 +1,168 @@
+import PaiTrialLog from './PaiTrialLog';
+import TrialLog from './TrialLog';
+import MessageInfo from '../MessageInfo';
+import PanelMonacoEditor from '../PanelMonacoEditor';
+import '@style/experiment/overview/overview.scss';
+
+/**
+ *  netron URL must be synchronized with ts/nni_manager/rest_server/index.ts`.
+ *  Remember to update it if the value is changed or this file is moved.
+ **/
+
+const OpenRow = (props): any => {
+    const [typeInfo, setTypeInfo] = useState('');
+    const [info, setInfo] = useState('');
+    const [isHidenInfo, setHideninfo] = useState(true);
+    const [showRetiaParamPanel, setShowRetiaparamPanel] = useState(false);
+    const trialId = props.trialId;
+    const trial = TRIALS.getTrial(trialId);
+    const logPathRow = trial.info.logPath || "This trial's log path is not available.";
+    const originParameters = trial.description.parameters;
+    const hasVisualHyperParams = RETIARIIPARAMETERS in originParameters;
+
+    const hideMessageInfo = (): void => {
+        setHideninfo(true);
+    };
+
+    const hideRetiaParam = (): void => {
+        setShowRetiaparamPanel(false);
+    };
+
+    const isshowRetiaParamPanel = (): void => {
+        setShowRetiaparamPanel(true);
+    };
+
+    /**
+     * info: message content
+     * typeInfo: message type: success | error...
+     * continuousTime: show time, 2000ms
+     */
+    const getCopyStatus = (info: string, typeInfo: string): void => {
+        setTypeInfo(typeInfo);
+        setInfo(info);
+        setHideninfo(false);
+        setTimeout(hideMessageInfo, 2000);
+    };
+
+    const copyParams = (trial: Trial): void => {
+        // get copy parameters
+        const params = JSON.stringify(reformatRetiariiParameter(trial.description.parameters as any), null, 4);
+        if (copy.default(params)) {
+            getCopyStatus('Successfully copy parameters to clipboard in form of python dict !', 'success');
+        } else {
+            getCopyStatus('Failed !', 'error');
+        }
+    };
+
+    const openTrialLog = (filename: string): void => {
+        window.open(`${MANAGER_IP}/trial-file/${props.trialId}/${filename}`);
+    };
+
+    const openModelOnnx = (): void => {
+        // TODO: netron might need prefix.
+        window.open(`/netron/index.html?url=${MANAGER_IP}/trial-file/${props.trialId}/model.onnx`);
+    };
+
+    return (
+        <Stack className='openRow'>
+            <Stack className='openRowContent'>
+                <Pivot>
+                    <PivotItem headerText='Parameters' key='1' itemIcon='TestParameter'>
+                        {trial.info.hyperParameters !== undefined ? (
+                            <Stack className='description'>
+                                <Stack className='bgHyper'>
+                                    <JSONTree
+                                        hideRoot={true}
+                                        shouldExpandNode={(): boolean => true} // default expandNode
```

Comment:
Maybe `() => true` is enough? Looks like it can auto-infer return type.

Reply:
yes! could be delete